### PR TITLE
Mbarba/logging gff

### PR DESCRIPTION
**Pull request by MatBarba**
Use the ensembl-py logging method to process GFF3 files
By default, nextflow creates a log_file

### CODE REVIEW

File: src/python/ensembl/io/genomio/gff3/process.py [hunk 1]
```diff
@@ -287,7 +287,7 @@ def simpler_gff3(self, in_gff_path: PathLike) -> None:
                     else:
                         fail_types["gene=" + feat.type] = 1
                         message = f"Unsupported feature type: {feat.type} (for {feat.id})"
-                        print(message)
+                        logging.warning(message)
```
Comment (JAlvarezJarreta) [inline src/python/ensembl/io/genomio/gff3/process.py:290]:
```suggestion
                        logging.warning(f"Unsupported feature type: {feat.type} (for {feat.id})")
```

File: src/python/ensembl/io/genomio/gff3/process.py [hunk 2]
```diff
@@ -549,7 +553,7 @@ def transcript_gene(self, ncrna: SeqFeature) -> SeqFeature:
     def cds_gene(self, cds: SeqFeature) -> SeqFeature:
         """Returns a gene created for a lone CDS."""
 
-        print("Put the lone CDS in gene-mRNA parent features")
+        logging.info("Put the lone CDS in gene-mRNA parent features")
```
Comment (JAlvarezJarreta) [inline src/python/ensembl/io/genomio/gff3/process.py:556]: Do we want to add some more information about what CDS is it about?

File: src/python/ensembl/io/genomio/gff3/process.py [hunk 3]
```diff
@@ -828,22 +832,22 @@ def valid_id(self, name: str) -> bool:
 
         # Trna (from tRNAscan)
         if re.search(r"^Trna", name):
-            print(f"Stable id is a Trna from tRNA-scan: {name}")
+            logging.debug(f"Stable id is a Trna from tRNA-scan: {name}")
```
Comment (JAlvarezJarreta) [inline src/python/ensembl/io/genomio/gff3/process.py:835]:
```suggestion
            logging.debug(f"Stable ID is a tRNA from tRNA-scan: {name}")
```

File: src/python/ensembl/io/genomio/gff3/process.py [hunk 4]
```diff
@@ -935,9 +939,12 @@ def main() -> None:
         default=Path("functional_annotation.json"),
         help="Output functional annotation JSON file",
     )
+    parser.add_log_arguments(add_log_file=True)
     args = parser.parse_args()
+    init_logging(args.log_level, args.log_file, args.log_file_level)
 
     # Merge multiline gene features in a separate file
+    logging.info("Check for genes to merge...")
```
Comment (JAlvarezJarreta) [inline src/python/ensembl/io/genomio/gff3/process.py:947]:
```suggestion
    logging.info("Checking for genes to merge...")
```

**Review by JAlvarezJarreta — APPROVED** [review]

File: src/python/ensembl/io/genomio/gff3/process.py (hunk 2 above)
Comment (MatBarba) [reply src/python/ensembl/io/genomio/gff3/process.py]: Yes; done

File: src/python/ensembl/io/genomio/gff3/process.py [hunk 5]
```diff
@@ -441,7 +444,7 @@ def _normalize_transcripts(self, gene: SeqFeature, fail_types) -> SeqFeature:
                 message = (
                     f"Unrecognized transcript type: {transcript.type}" f" for {transcript.id} ({gene.id})"
                 )
-                print(message)
+                logging.warning(message)
```
Comment (JAlvarezJarreta) [inline src/python/ensembl/io/genomio/gff3/process.py:447]:
```suggestion
                logging.warning(
                    f"Unrecognized transcript type: {transcript.type}" f" for {transcript.id} ({gene.id})"
                )
```
And remove the previous `message` declaration?

File: src/python/ensembl/io/genomio/gff3/process.py [hunk 6]
```diff
@@ -515,7 +518,7 @@ def _normalize_transcript_subfeatures(
                     f"Unrecognized exon type for {feat.type}: {feat.id}"
                     f" (for transcript {transcript.id} of type {transcript.type})"
                 )
-                print(message)
+                logging.warning(message)
```
Comment (JAlvarezJarreta) [inline src/python/ensembl/io/genomio/gff3/process.py:521]: Same suggestion as before:
```suggestion
                logging.warning((
                    f"Unrecognized exon type for {feat.type}: {feat.id}"
                    f" (for transcript {transcript.id} of type {transcript.type})"
                ))
```

File: src/python/ensembl/io/genomio/gff3/process.py [hunk 7]
```diff
@@ -935,9 +938,12 @@ def main() -> None:
         default=Path("functional_annotation.json"),
         help="Output functional annotation JSON file",
     )
+    parser.add_log_arguments(add_log_file=True)
     args = parser.parse_args()
+    init_logging(args.log_level, args.log_file, args.log_file_level)
```
Comment (JAlvarezJarreta) [inline src/python/ensembl/io/genomio/gff3/process.py:943]:
```suggestion
    init_logging_with_args(args)
```

File: src/python/ensembl/io/genomio/gff3/process.py [hunk 8]
```diff
@@ -35,6 +36,7 @@
 
 from ensembl.io.genomio.gff3.extract_annotation import FunctionalAnnotations
 from ensembl.utils.argparse import ArgumentParser
+from ensembl.utils.logging import init_logging
```
Comment (JAlvarezJarreta) [inline src/python/ensembl/io/genomio/gff3/process.py:39]:
```suggestion
from ensembl.utils.logging import init_logging_with_args
```

**Review by JAlvarezJarreta — APPROVED** [review]
Couple of minor edits, but looks good :+1:

**Pull request merged**